### ISSUE TITLE
Box Build method inside a Virtual Machine: cannot write directly to shared host folder

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-installer#103]
When using kiwi's box build method inside a Virtual Machine (using Virtualbox), it seems that specifying an output directory that points to a shared folder (between guest and host OS) is not possible.
This constraint applies to the box build method using the classic VM approach as well as using Vagrant in combination with the box build.

It might be related to this, but I am not sure:

http://osinside.github.io/kiwi/plugins/self_contained.html#sharing-backends

Possibly the default method that box build uses would need to be switched explicitly to `--sshfs-sharing` or `--virtiofs-sharing`. The second switch seems more cumbersome, while the third one doesn't require additional config but is experimental at this time. I have not tried either of those, but can (at least the third one).

Or an update to the documentation that just highlights the fact that without more complex command lines, the easiest will be to contain the build within the VM itself and then manually copy it out to a shared folder if so desired (meaning enough disk space needs to be available to house the iso and related files during processing).

Note:
The vagrant update I am working on would switch to using the build box approach and essentially just add the step of copying from within the VM into the shared directory at the end of processing.

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for reporting this. Much appreciated. I'm currently favouring your:

> The vagrant update I am working on would switch to using the build box approach and essentially just add the step of copying from within the VM into the shared directory at the end of processing.

brute force approach. This is more a 'keeping it simple' approach if that can even be a thing with a virtual host mount within a VM building an OS installer using another VM 'box'. This way we depend on less 'special' options and go more with the flow of what works now and is likely less brittle to the changing nature of whatever tec is used to present the shared host fs.

As per the doc entry we could simply state this limitation of the boxbuild method with the associated copy requirement as a limitation, linking to your sited kiwi-ng doc section. In the scope of things the extra space requirement is not that much. I have in my notes to leave 5 to 6 GB of free space for a fresh installer build. I think it took around 3.5 GB when I last measured it. Plus we already indicate within the readme "... 15 GB free space ...".

**Comment by Hooverdan96** [comment]
Sounds good. I will create a PR for updating the documentation. I will do that as soon as I have worked out (or abandoned :) ) the Vagrant approach in the next few days. If I figure it out, I will create a separate issue/PR for that naturally, but that should inform this Issue as well.

**Comment by Hooverdan96** [comment]
Feel free to suggest other ways to document this.

**Comment by phillxnet** [comment]
@Hooverdan96 You pr is great. Thanks. I've now reviewed with a minor change suggested.

Re moving to semantic formatting. We already do this in our website material and post my review I realised we haven't gone that way yet here in the readme.md. But lets see how it looks in GitHub anyway (as a mix) and we can later move the entire readme whole-sale if it renders OK. I think it makes for easier edits/additions as it provides better 'chunking' and avoids messing about with more formatting that need be. And helps to keep line length down of course. I also like to see the overview of sentence, major part there of, length. See what you think.
